bootstrap.sh

```diff
@@ -1,10 +1,77 @@
 #!/bin/bash
 
-set -ueo pipefail
-
 echo "/dev/disk/by-id/google-elasticpd /var/lib/elasticsearch ext4 defaults 0 0" >> /etc/fstab
 lsblk -o label /dev/disk/by-id/google-elasticpd | grep -q elastic-data || mkfs.ext4 -L elastic-data /dev/disk/by-id/google-elasticpd
 mount /var/lib/elasticsearch
 chown elasticsearch:elasticsearch /var/lib/elasticsearch
 
+
+if ! grep 'elasticsearch soft memlock unlimited' /etc/security/limits.conf; then
+
+(
+cat <<ADDTEXT
+elasticsearch soft memlock unlimited
+elasticsearch hard memlock unlimited
+ADDTEXT
+) >> /etc/security/limits.conf;
+
+fi
+
+su elasticsearch -s /bin/bash -c '/usr/share/elasticsearch/bin/elasticsearch | tee /tmp/elasticsearch_bootstrap_start.log >/dev/full' &
+
+echo "waiting for ES7 bootstrap to start"
+while true; do
+  grep "starting GCE discovery service" /tmp/elasticsearch_bootstrap_start.log && break;
```

Comment:
Tohle bych přepsala na něco jako:

```bash
while grep -v "starting GCE discovery service" /tmp/elasticsearch_bootstrap_start.log; do
  sleep 10
done
```

A asi bych se podívala, jestli na ten start neumíme počkat lepším způsobem než grepováním logu.

Reply:
Nedalo by se použít třeba https://www.elastic.co/guide/en/elasticsearch/reference/7.7/cat-health.html?

Reply:
V týhle části ještě ne, protože se to odmítne zapnout a api ještě nic nevrací. To když už, tak na řádku 85